pkg/multicluster/cluster_management.go

```diff
@@ -84,17 +84,25 @@ func (clusterConfig *KubeClusterConfig) Validate() error {
 	return nil
 }
 
-// RegisterByVelaSecret create cluster secrets for KubeVela to use
-func (clusterConfig *KubeClusterConfig) RegisterByVelaSecret(ctx context.Context, cli client.Client) error {
-	if err := ensureClusterNotExists(ctx, cli, clusterConfig.ClusterName); err != nil {
-		return errors.Wrapf(err, "cannot use cluster name %s", clusterConfig.ClusterName)
+// PostRegistration try to create namespace after cluster registered. If failed, cluster will be unregistered.
+func (clusterConfig *KubeClusterConfig) PostRegistration(ctx context.Context, cli client.Client) error {
+	if clusterConfig.CreateNamespace != "" {
```

Comment:
To keep the body simple, you can check whether CreateNamespace is empty and return nil  :)

Reply:
Good idea!